pkg/flags/strings.go

```diff
@@ -25,25 +28,87 @@ func NewStringsFlag(valids ...string) *StringsFlag {
 	return &StringsFlag{Values: valids, val: valids[0]}
 }
 
-// StringsFlag implements the flag.Value interface.
+// StringsFlag implements the flag.Value and pflag.Value interfaces.
 type StringsFlag struct {
 	Values []string
 	val    string
 }
 
 // Set verifies the argument to be a valid member of the allowed values
 // before setting the underlying flag value.
-func (ss *StringsFlag) Set(s string) error {
-	for _, v := range ss.Values {
+func (sf *StringsFlag) Set(s string) error {
+	for _, v := range sf.Values {
 		if s == v {
-			ss.val = s
+			sf.val = s
 			return nil
 		}
 	}
 	return errors.New("invalid value")
 }
 
 // String returns the set value (if any) of the StringsFlag
-func (ss *StringsFlag) String() string {
-	return ss.val
+func (sf *StringsFlag) String() string {
+	return sf.val
+}
+
+// Type returns the given type as string
+func (sf *StringsFlag) Type() string {
+	return "string"
+}
+
+// StringSliceFlag implements the flag.Value and pflag.Value interfaces.
+type StringSliceFlag struct {
+	Values []string
+	val    []string
+}
+
+// NewStringSliceFlag creates a new string slice flag for which any one of the given
+// strings is a valid value, and any other value is an error.
+func NewStringSliceFlag(valids ...string) *StringSliceFlag {
```

Comment:
suggest giving separate control over valid values and default value, instead of assuming `[]string{}`